### PR TITLE
Added unit tests for the functions in object.

### PR DESCRIPTION
**Pull request by EstFoisy**
#### What it does

The pull-request adds unit-tests for methods in `object.ts`.

It test functions from object file.

#### How to test

The CI should pass for the newly added tests.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/common/object.spec.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    describe('#isEmpty', () => {
+        it('should return "true" when it is Empty', () => {
+            const z = {};
+            expect(isEmpty(z)).to.equal(true);
+        });
+        it('should return "false" when not Empty', () => {
+            const z = { d: 5 };
+            expect(isEmpty(z)).to.equal(false);
+        });
+    });
```
Comment (colin-grant-work) [inline packages/core/src/common/object.spec.ts:109]: `isEmpty` checks two features of an object: that it has keys, and that its constructor is `Object`. These tests only confirm the behavior of one check. How can we check the `Object` constructor feature of `isEmpty`?

Comment (EstFoisy) [reply packages/core/src/common/object.spec.ts]: Hi @colin-grant-work , 
Thanks a lot for advising about the above: I have revised my units tests according to your comment.

File: packages/core/src/common/object.spec.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
+// *****************************************************************************
```
Comment (vince-fugnitto) [inline packages/core/src/common/object.spec.ts:1]: Given that the file is `objects.ts` the file should be renamed to `objects.spec.ts` (notice the missing `s`).

File: packages/core/src/common/object.spec.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            expect(notEmpty({ a: 1 })).to.equal(true);
+        });
+        it('should return "false" when empty', () => {
+            expect(notEmpty(undefined)).to.equal(false);
+        });
+    });
+
+    describe('#isEmpty', () => {
+        it('should return "true" when it is Empty', () => {
+            const obj = Object.create({});
```
Comment (vince-fugnitto) [inline packages/core/src/common/object.spec.ts:102]: I don't believe this tests the non-object `constructor`.


```bash
$ node
Welcome to Node.js v16.14.0.
Type ".help" for more information.
> a = Object.create({})
{}
> a.constructor
[Function: Object]
> a.constructor === Object
true

```

File: packages/core/src/common/object.spec.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            expect(notEmpty([])).to.equal(true);
+            expect(notEmpty({ a: 1 })).to.equal(true);
+        });
+        it('should return "false" when empty', () => {
+            expect(notEmpty(undefined)).to.equal(false);
+        });
+    });
+
+    describe('#isEmpty', () => {
+        it('should return "true" when it is Empty', () => {
```
Comment (vince-fugnitto) [inline packages/core/src/common/object.spec.ts:101]:
```suggestion
        it('should return "true" when it is empty', () => {
```

File: packages/core/src/common/object.spec.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            const obj = Object.create({});
+            expect(isEmpty(obj)).to.equal(true);
+        });
+        it('should return "false" when not Empty', () => {
```
Comment (vince-fugnitto) [inline packages/core/src/common/object.spec.ts:105]:
```suggestion
        it('should return "false" when not empty', () => {
```

**Review by vince-fugnitto — APPROVED** [review]
LGTM :+1:

**Pull request merged**